pkg/auth/providers/azure/config.go

```diff
@@ -24,17 +28,41 @@ func graphEndpointDeprecated(endpoint string) bool {
 }
 
 func updateAzureADConfig(c *v32.AzureADConfig) {
+	if isConfigGlobal(c) {
+		updateConfigForGlobal(c)
+	} else {
+		updateConfigForChina(c)
+	}
+}
+
+func isConfigGlobal(c *v32.AzureADConfig) bool {
+	return !strings.HasSuffix(c.GraphEndpoint, ".cn") &&
+		!strings.HasSuffix(c.GraphEndpoint, ".cn/")
+}
+
+func updateConfigForGlobal(c *v32.AzureADConfig) {
 	// Update the Graph Endpoint.
-	graphEndpointMigration := map[string]string{
-		globalAzureADGraphEndpoint: globalMSGraphEndpoint,
-		chinaAzureADGraphEndpoint:  chinaMSGraphEndpoint,
+	if c.GraphEndpoint == globalAzureADGraphEndpoint {
+		c.GraphEndpoint = globalMSGraphEndpoint
 	}
-	// This will be a no-op if the config already uses the new endpoint.
-	if newEndpoint, oldUsed := graphEndpointMigration[c.GraphEndpoint]; oldUsed {
-		c.GraphEndpoint = newEndpoint
+
+	// Update the Auth Endpoint and Token Endpoint.
+	c.AuthEndpoint = fmt.Sprintf("%s%s/oauth2/v2.0/authorize", c.Endpoint, c.TenantID)
+	c.TokenEndpoint = fmt.Sprintf("%s%s/oauth2/v2.0/token", c.Endpoint, c.TenantID)
```

Comment:
I agree, but I'd rather keep this specific to each region, even if there is duplication, because Microsoft may change how endpoints must be specified, and, overall, I'd keep this endpoint manipulation fully separate and encapsulated for each region, because they are not exactly the same (Endpoint for login, for example, is not needed for Global but needed for China).

Reply:
I can get behind that reasoning 👍